definitions/tools/sequence_align_and_tag.cwl

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "align with bwa_mem and tag"
+
+baseCommand: ["/bin/bash", "sequence_alignment_helper.sh"]
+requirements:
+    - class: SchemaDefRequirement
+      types:
+          - $import: ../types/sequence_data.yml
+    - class: ResourceRequirement
+      coresMin: 8
+      ramMin: 20000
+    - class: DockerRequirement
+      dockerPull: "mgibio/alignment_helper-cwl:1.0.0"
+    - class: InlineJavascriptRequirement
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'sequence_alignment_helper.sh'
+        entry: |
+            set -o pipefail
+            set -o errexit
+            set -o nounset
+
+            while getopts "b:?1:?2:?g:r:n:" opt; do
+                case "$opt" in
+                    b)
+                        MODE=bam
+                        BAM="$OPTARG"
+                        ;;
+                    1)
+                        MODE=fastq
+                        FASTQ1="$OPTARG"
+                        ;;
+                    2)  
+                        MODE=fastq
+                        FASTQ2="$OPTARG"
+                        ;;
+                    g)
+                        READGROUP="$OPTARG"
+                        ;;
+                    r)
+                        REFERENCE="$OPTARG"
+                        ;;
+                    n)
+                        NTHREADS="$OPTARG"
+                        ;;
+                esac
+            done
+
+            if [[ "$MODE" == 'fastq' ]]; then
+                /usr/local/bin/bwa mem -K 100000000 -t "$NTHREADS" -Y -p -R "$READGROUP" "$REFERENCE" "$FASTQ1" "$FASTQ2" | /usr/local/bin/samblaster -a --addMateTags | /opt/samtools/bin/samtools view -b -S /dev/stdin
+            fi
```

Comment:
Should this be an else-if in case someone defined BAM and FASTQ inputs? Then the FASTQ would have precedence over the BAM (since it's more efficient).

Reply:
Only one of the two will ever happen.  In this bash script, whichever parameter is encountered last will determine the `$MODE`.

Further, the CWL type definition uses "exclusive" parameters, so it will error if both BAM and FASTQs are supplied.  (The two FASTQ parameters are also dependent so both must be supplied if one is.  See also [this user guide page](https://www.commonwl.org/user_guide/11-records/) for more details.)